From d7f96a49a4887c2f829b5c6671d794e1ca2d3568 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:01:11 -0400
Subject: [infiniband] mlx4: fix unitialised variable is_mcast

Message-id: <cb754d62744d2f3b4a64ded45df20831c713f9d1.1410425016.git.avadai@redhat.com>
Patchwork-id: 93684
O-Subject: [Patch RHEL7.1 V1 135/197] IB/mlx4: fix unitialised variable is_mcast
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Colin Ian King <colin.king@canonical.com>

Commit 297e0dad7206 ("IB/mlx4: Handle Ethernet L2 parameters for IP
based GID addressing") introduced a bug where is_mcast is now no
longer initialized on the non-multicast condition and so it can be
any random value from the stack.  This issue was detected by cppcheck:

    [drivers/infiniband/hw/mlx4/ah.c:103]: (error) Uninitialized
      variable: is_mcast

Simple fix is to initialise is_mcast to zero.

Signed-off-by: Colin Ian King <colin.king@canonical.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit bfdfcfee3c9281e9cd28c0b08235aba1762504a6)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/mlx4/ah.c b/drivers/infiniband/hw/mlx4/ah.c
index 170dca6..2d8c339 100644
--- a/drivers/infiniband/hw/mlx4/ah.c
+++ b/drivers/infiniband/hw/mlx4/ah.c
@@ -73,7 +73,7 @@ static struct ib_ah *create_iboe_ah(struct ib_pd *pd, struct ib_ah_attr *ah_attr
 {
  struct mlx4_ib_dev *ibdev = to_mdev(pd->device);
  struct mlx4_dev *dev = ibdev->dev;
- int is_mcast;
+ int is_mcast = 0;
  struct in6_addr in6;
  u16 vlan_tag;
 
-- 
1.7.1